### ISSUE TITLE
Could not start server due to permission denied on sibling folder

### ISSUE THREAD

**Issue by damienrg** [bash-lsp/bash-language-server#139]
Starting `bash-language-server` on a file fails when the user has no permissions to read a folder next to the file.

The exact error is:

```
glob error { [Error: EACCES: permission denied, scandir '/home/foo/folder-without-permissions']
  errno: -13,
  code: 'EACCES',
  syscall: 'scandir',
  path: '/home/foo/folder-without-permissions' }
```

and this is due to:
https://github.com/mads-hartmann/bash-language-server/blob/2d3528d670277858aaf52a3ebe6577beea524876/server/src/analyser.ts#L47-L49

I think that you should show a warning instead of triggering an error and that recursing 
in all directories and sub-directories is a bit aggressive.

**Comment by skovhus** [comment]
@damienrg agree. Thanks for reporting this.

Let me know if you would like to do a pull request.

**Comment by damienrg** [comment]
I can do it but I need to know what I should do:

1. replace the error by a warning
2. remove the recursive glob (could break things)
3. only search in the current directory  (could break things)
4. check if  the `rootPath` contains a specific file that should contain the list of files to consider. In this case we do not do the recursive glob.

**Comment by nhayden** [comment]
Even just a better diagnostic message would make a big difference. Like `bash-language-server` is trying to scan sibling directories to provide better suggestions, but doesn't have read permissions.

**Comment by skovhus** [comment]
@damienrg @nhayden I believe this has been fixed. Please reopen if this is not the case.

Thanks.

**Comment by damienrg** [comment]
For information, the commit that fixes this issue is 52bd910fd307da05d8d8fe2fa43d3af990d79da3 by providing customization points:

- VSCode users can modify `bashIde.globPattern`
- Other users can set `GLOB_PATTERN` environment variable

The default pattern is `**/*@(.sh|.inc|.bash|.command)`.
